### ISSUE TITLE
Does the code-specific depth gain grow over training, or is it structural?

### ISSUE THREAD

**Issue by MatthewLacerda2** [MatthewLacerda2/TinyRefinementModel#242]
Pre-registered as `experiments/depth/specs/242-depth-gain-over-training.toml`, committed before any results.

## The fork

Depth recurrence buys the champion **+0.003454 nats on code** (t=+15.8, n=204,380) and nothing measurable on prose or math. That is the *final* checkpoint only. Two readings fit it and they mean different things:

- **LEARNED** — the model gradually learns to use the refine loop on structured input. The gain should be small or absent early and grow.
- **STRUCTURAL** — looping a causal block simply helps on code-shaped input, for reasons unrelated to training. The gain should be roughly as large at 0.66B tokens as at 3.99B.

The second would make "the model learned where depth pays" wrong, and would redirect the question from the training recipe to the architecture. Worth resolving before anything is written up.

**Prediction registered: LEARNED**, on the strength of the gate observations in #225 — the refine loop writes to 6 of 960 channels on prose and up to 447 on code, a routing the model had to arrive at, since `gate_bias` initialises balanced at sigmoid(0).

## Rungs

What survived retention: **0.66B** (step 647167), **3.20B** (3121279), **3.99B** (champion). The 1.45B checkpoints are labelled `collapse_onset`/`collapsed` and are excluded as damaged rather than quietly averaged in.

The 3.20B rung returns all-NaN logits on ~10% of code documents (#229). It is carried for the table but **deliberately excluded from every criterion** — a checkpoint failing on a tenth of the corpus cannot anchor a trend. `instruments/paired.py` drops and counts non-finite positions rather than averaging over them, so that rung reports on fewer positions and says so.

## Note on sigma

Seeds vary the held-out **document sample**, not model init — the checkpoints are fixed and the forward pass deterministic, so document sampling is the only honest source of variation. This is *not* #17's seed-to-seed training noise floor (σ ≈ 0.03 nats), which is the wrong bar by two orders of magnitude and the specific mistake `paired.py` exists to prevent.

**Comment by MatthewLacerda2** [comment]
## Verdict: INCONCLUSIVE — and my registered prediction was wrong in the way that mattered

```
verdict: INCONCLUSIVE — keep criteria not met: prose_stays_neutral
  gain_grows_on_code           [beats 2σ]  → met
  prose_stays_neutral          [within 2σ] → NOT met
  gain_was_there_from_the_start[within 2σ] → NOT met
```

Recorded results, paired depth-8-vs-depth-1 mean gain in nats, 4 document samples per arm:

```
                    0.66B              3.20B            3.99B
codeparrot     -6.2523 … -6.1118   +0.0055 … +0.0037  +0.0019 … +0.0011
fineweb-edu    -5.2498 … -5.3123   -0.0005 … -0.0003  -0.0001 … -0.0002
```

I registered **LEARNED** — "the model gradually learns to use the refine loop on structured
input". The direction is right and the mechanism is not. Absolute CE on codeparrot makes it
plain (uniform-random over this vocab is 10.82 nats):

```
                     d1      d2      d4      d8
0.66B  647167       2.200   7.892   8.325   8.169
3.99B  champion     1.528   1.528   1.529   1.529
```

**One refine pass is fine. The second costs 5.7 nats.** At 0.66B the loop does not fail to
help — it actively destroys the prediction, landing closer to random than to competent. By
3.99B, eight passes cost nothing at all: flat to three decimal places.

## What that reframes

The model is not learning to **use** depth. It is learning to **survive** it.

That re-reads #225's gate measurement, which I had the wrong way round. The refine loop writing
to 6 of 960 channels on prose is not a routing decision about where depth pays — it is the
model switching the loop off, because the loop was wrecking its output. On code it suppresses
less (up to 447 channels) and ends up marginally ahead. The +0.0035 nats on code is not depth
being exploited; it is the residue left after suppression.

It also explains `gate_bias`, which exists as a workaround for "the depth-8 collapse where
balanced steps compound and diverge". That collapse is not a toy-scale curiosity. It is what
this architecture does at 0.66B on real data, and taming it is apparently what a large part of
training goes on doing.

## Why INCONCLUSIVE is the right verdict, not a technicality

`prose_stays_neutral` failed because prose went from **−5.25 to −0.0001**. I wrote that
criterion assuming prose would be flat throughout and only code would move. It was not flat;
it was catastrophic and then recovered. The criterion caught an assumption I did not know I
was making, which is what pre-registration is for.

Two honest problems with the spec itself, both mine:

- **The sigma framing is meaningless at this separation.** "+124σ" and "+283σ" are artefacts of
  comparing quantities three orders of magnitude apart with a sampling sigma of ~0.0004.
  Criteria phrased in sigmas quietly assume the arms are comparable in magnitude. Worth a note
  in the spec format doc.
- **Effect size varies by corpus region more than the within-sample error suggests.** This
  sweep put the champion's code gain at +0.0011…+0.0019 across four samples; an earlier run on
  a different slice (400 rows, skip 1.5M) gave +0.0035 with se 0.0002. Tight within a slice,
  3× apart between slices. Document heterogeneity is the dominant uncertainty and the reported
  standard error understates it.

## What I am NOT claiming

The d2 collapse was **not pre-registered** — it fell out of the results. So it is a hypothesis,
not a finding, however stark the numbers. It needs its own spec: more rungs between 0.66B and
3.20B to date when the loop stops being destructive, the per-depth curve at each, and prose as
the control throughout. Only two usable rungs survived retention, which is itself a lesson for
checkpoint policy (#187).

**Comment by MatthewLacerda2** [comment]
Done: the spec experiments/depth/specs/242-depth-gain-over-training.toml was run and its result is cited in docs/findings/2026-09-12-depth-recurrence-is-suppressed-not-exploited.md.
